src/core/EdFi.DataManagementService.Core/Middleware/ValidateImmutableIdentityMiddleware.cs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Response;
+using EdFi.DataManagementService.Core.Validation;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.Middleware
+{
+    internal class ValidateImmutableIdentityMiddleware(ILogger _logger, IImmutableIdentityValidator _validator) : IPipelineStep
```

Comment:
This is good stuff, but the name isn't quite right. "Immutable identity" is about not allowing someone to change the identity fields (e.g. schoolId) on an existing document, which we detect using ReferentialIds. This about mismatched DocumentUuids. Maybe "ValidateMatchingDocumentUuidsMiddleware"?